tests/test_requests_resource.py

```diff
@@ -0,0 +1,248 @@
+            content_type=("application/json")
+        )
+
+        self.assertEqual(response.status_code, 201)
+        response = self.client.get('/api/v1/users/requests/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_get_a_request(self):
+        """ Test a resource can be successfully retrived """
+        # self.register_and_authenticate_user()
+
+        response = self.client.post(
+            '/api/v1/users/requests/',
+            data=json.dumps(self.data["dummy_request"]),
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 201)
+        # created_resource = json.loads(response.data)
+        response = self.client.get('/api/v1/users/request/1/')
+        self.assertEqual(response.status_code, 200)
+
+    def test_post_a_request(self):
+        """ Test a resource can be successfully created"""
+        response = self.client.post(
+            '/api/v1/users/requests/',
+            data=json.dumps(self.data["dummy_request"]),
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 201)
+
+    def tests_empty_dict(self):
+        response = self.client.post(
+            'api/v1/users/requests/',
+            data={},
+            content_type=("application/json")
+        )
+
+        self.assertEqual(response.status_code, 400)
+
+    def tests_empty_list(self):
+        response = self.client.post(
+            'api/v1/users/requests/',
+            data=[],
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def tests_empty_string(self):
+        response = self.client.post(
+            'api/v1/users/requests/',
+            data="",
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def tests_empty_tuple(self):
+        response = self.client.post(
+            'api/v1/users/requests/',
+            data=(),
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def tests_bad_data(self):
+        response = self.client.post(
+            'api/v1/users/requests/',
+            data={
+                "request_id": 1,
+                'title': '',
+                "location": 234
+            },
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 400)
+
+    def test_update_a_request(self):
+        """ Test a resource can be successfully updated """
+        response = self.client.post(
+            '/api/v1/users/requests/',
+            data=json.dumps(self.data["dummy_request"]),
+            content_type=("application/json")
+        )
+        self.assertEqual(response.status_code, 201)
+
+        response = self.client.put(
+            '/api/v1/users/request/1/', data=json.dumps(dict(location="New title")), content_type=("application/json")
```

Comment:
Make this line shorter like the rest. This increases readability and remember PEP 8 styling :)